### DISCUSSION TITLE
ETCD store for Filer and client cert authentication

### DISCUSSION THREAD

**Discussion by WhiteSash** [seaweedfs/seaweedfs#2089] [Q&A]
First of all, thank you very much for this project!
In my case, there is already a cluster of etcd with configured authentication for a client certificate. I would like to use the same cluster without compromising its security. Other software that I use has this kind of support. I have not found a description of such functionality in the help or wiki. 
Is there a way to specify for use of the client certificate in the filler configuration when stored in the etcd?
If at the moment it does not work, what are the plans to add such functionality?

**Comment by chrislusf** [comment]
Supporting it should be trivial. But no good way to test it. You can do it yourself. You just need to change this piece to add the support:

https://github.com/chrislusf/seaweedfs/blob/master/weed/filer/etcd/etcd_store.go#L48

And add some plumbing here:
https://github.com/chrislusf/seaweedfs/blob/master/weed/filer/etcd/etcd_store.go#L34
https://github.com/chrislusf/seaweedfs/blob/master/weed/command/scaffold.go#L252

Welcome to send a PR.
